### ISSUE TITLE
[BUG] ox-odt fails for org-id links (e.g., from org-roam v2) 9.5.2 / Also it should convert `org` links to `odt` links, if requested

### ISSUE THREAD

**Issue by kjambunathan** [kjambunathan/org-mode-ox-odt#184]
Source [[BUG] ox-odt fails for org-id links (e.g., from org-roam v2) 9.5.2](https://lists.gnu.org/archive/html/emacs-orgmode/2022-05/msg00225.html)

# [BUG] ox-odt fails for org-id links (e.g., from org-roam v2) 9.5.2

When exporting an org-file with org-id based links  to odt, I get the error error "FIXME: Unable to resolve <filename>", where <filename> is the filename, the org-id-link was pointing to.

Org-id links (example below) are the default for org-roam (v2).

[[id:5222a0dd-5084-41bd-b61d-21dcb3290187][Note title]]

The issue lies within ox-odt.el in the org-odt-link function. There, 'destination' takes the value of the file name of the id-link target. Therefore, (org-element-type destination) gets evaluated to 'plain-text, which is not resolvable by 'org-odt-link--infer-description', thus it raises the error. 

I worked around this issue by adding a check for plain-text, but would like to raise this issue here to find proper ideas on how to fix this.

```lisp
      (cond
              ;; workaround
     ((eq (org-element-type destination) 'plain-text)
      (format "<text:a xlink:type=\"simple\" xlink:href="">               destination
               desc
               ))
               ;; workaround end
     ((not label-reference)
      (org-odt-link--infer-description destination info))
```

Thank you,
Moritz

Emacs  : GNU Emacs 27.2 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.24.30, cairo version 1.16.0)
Package: Org mode version 9.5.2 (9.5.2-gfbff08 @ /home/moritz/.emacs.d/elpa/27.2/develop/org-9.5.2/)

**Comment by fleimgruber** [comment]
Thanks for bringing this up, ran into the same issue!

But your workaround does not work as is, I guess you wanted something like this?

```emacs-lisp
...
(cond
  ;; workaround
  ((eq (org-element-type destination) 'plain-text)
   (format "<text:a xlink:type=\"simple\" xlink:href=\"%s\">%s</text:a>"
           destination
           desc))
  ;; workaround end
  ((not label-reference)
  ...

```

**Comment by kjambunathan** [comment]
Test case 

`main.org`

```
:PROPERTIES:
:ID:       450948ae-c58b-43ca-8208-ef6c4e158b8a
:END:

 #+begin_src emacs-lisp
 (setq org-roam-directory (file-truename "/home/kjambunathan/src/org-mode-ox-odt/testing/examples/odt/orgroam/"))
 (org-roam-db-autosync-mode)
 #+end_src

 #+RESULTS:
 : t

* Main content
:PROPERTIES:
:ID:       9ec20896-56d2-41be-ba27-d8f959853afd
:END:

[[id:a3455ed6-194c-44e4-80ff-64700e7bfbd9][Other]]


```

`other.org`

```
:PROPERTIES:
:ID:       807c6ecf-f818-48cd-8312-3a2d3feb0d3d
:END:

* Other
:PROPERTIES:
:ID:       a3455ed6-194c-44e4-80ff-64700e7bfbd9
:END:

```

**Comment by kjambunathan** [comment]
This is how the output should look like 


```xml
<text:a text:style-name="Internet_20_link"
        text:visited-style-name="Visited_20_Internet_20_Link"
        xlink:href="../other.odt#ID-a3455ed6-194c-44e4-80ff-64700e7bfbd9"
        xlink:type="simple">other.odt</text:a>
```

**Comment by kjambunathan** [comment]
This issue is fixed in `ox-odt-9.5.3.464`.  

This fix is now available in my fork.  FWIW, the ODT exporter in upstream Emacs and org-mode is for all practical puposes dead.  So, if you are interested in bug fixes and enhancements, use this fork.   For more information see [How to install the *enhanced* ODT Exporter](https://github.com/kjambunathan/org-mode-ox-odt#installation).
